src/staking-router-modules/staking-router.service.ts

```diff
@@ -0,0 +1,123 @@
+import { EntityManager } from '@mikro-orm/knex';
+import { Inject, Injectable, LoggerService, NotFoundException } from '@nestjs/common';
+import { LOGGER_PROVIDER } from '@lido-nestjs/logger';
+import { ModuleRef } from '@nestjs/core';
+import { StakingModuleInterface } from './interfaces/staking-module.interface';
+import { httpExceptionTooEarlyResp } from '../http/common/entities/http-exceptions';
+import { ELBlockSnapshot } from '../http/common/entities';
+import { config } from './staking-module-impl-config';
+import { IsolationLevel } from '@mikro-orm/core';
+import { SrModuleEntity } from '../storage/sr-module.entity';
+import { SRModuleStorageService } from '../storage/sr-module.storage';
+import { ElMetaStorageService } from '../storage/el-meta.storage';
+import { ElMetaEntity } from '../storage/el-meta.entity';
+
+@Injectable()
+export class StakingRouterService {
+  constructor(
+    @Inject(LOGGER_PROVIDER) protected readonly logger: LoggerService,
+    private readonly moduleRef: ModuleRef,
+    protected readonly entityManager: EntityManager,
+    protected readonly srModulesStorage: SRModuleStorageService,
+    protected readonly elMetaStorage: ElMetaStorageService,
+  ) {}
+
+  /**
+   * Method for reading staking modules from database
+   * @returns Staking module list from database
+   */
+  public async getStakingModules(): Promise<SrModuleEntity[]> {
+    const srModules = await this.srModulesStorage.findAll();
+    return srModules;
+  }
+
+  /**
+   * Method for reading staking module from database by module id
+   * @param moduleId - id or address of staking module
+   * @returns Staking module from database
+   */
+  public async getStakingModule(moduleId: string | number): Promise<SrModuleEntity | null> {
+    if (typeof moduleId == 'string') {
+      return await this.srModulesStorage.findOneByContractAddress(moduleId);
+    }
+
+    return await this.srModulesStorage.findOneById(moduleId);
+  }
+
+  public getStakingRouterModuleImpl(moduleType: string): StakingModuleInterface {
+    const impl = config[moduleType];
+    const moduleInstance = this.moduleRef.get<StakingModuleInterface>(impl);
+    return moduleInstance;
+  }
+
+  /**
+   * Execution layer meta that is stored in database
+   */
+  public async getElBlockSnapshot(): Promise<ElMetaEntity | null> {
+    return await this.elMetaStorage.get();
+  }
+
+  /**
+   * Helper method for getting staking module list and execution layer meta
+   * @returns Staking modules list and execution layer meta
+   */
+  public async getStakingModulesAndMeta(): Promise<{
+    stakingModules: SrModuleEntity[];
+    elBlockSnapshot: ELBlockSnapshot;
+  }> {
+    const { stakingModules, elBlockSnapshot } = await this.entityManager.transactional(
+      async () => {
+        const stakingModules = await this.getStakingModules();
+
+        if (stakingModules.length === 0) {
```

Comment:
In the next iteration, I believe it would be better to move this logic to the middleware